package.json

```diff
@@ -83,7 +83,7 @@
     "make-generator-function": "^1.1.0",
     "semver": "^5.5.0",
     "seq": "0.3.5",
-    "tap": "^10.7.2",
+    "tap": "^12.5.3",
```

Comment:
What are the breaking changes here?

Reply:
npm audit kicks off a notice of 15 vulnerabilities, see
```
infintesimal@hishost:~/bw$ npm audit --parseable
install	handlebars	high	npm install --save-dev tap@12.5.3	Prototype Pollution	https://npmjs.com/advisories/755	tap>nyc>istanbul-reports>handlebars	Y
install	hoek	moderate	npm install --save-dev tap@12.5.3	Prototype Pollution	https://npmjs.com/advisories/566	tap>coveralls>request>hawk>boom>hoek	Y
install	hoek	moderate	npm install --save-dev tap@12.5.3	Prototype Pollution	https://npmjs.com/advisories/566	tap>coveralls>request>hawk>cryptiles>boom>hoek	Y
install	hoek	moderate	npm install --save-dev tap@12.5.3	Prototype Pollution	https://npmjs.com/advisories/566	tap>coveralls>request>hawk>hoek	Y
install	hoek	moderate	npm install --save-dev tap@12.5.3	Prototype Pollution	https://npmjs.com/advisories/566	tap>coveralls>request>hawk>sntp>hoek	Y
install	tunnel-agent	moderate	npm install --save-dev tap@12.5.3	Memory Exposure	https://npmjs.com/advisories/598	tap>coveralls>request>tunnel-agent	Y
install	lodash	moderate	npm install --save-dev tap@12.5.3	Prototype Pollution	https://npmjs.com/advisories/782	tap>nyc>istanbul-lib-instrument>babel-generator>babel-types>lodash	Y
install	lodash	moderate	npm install --save-dev tap@12.5.3	Prototype Pollution	https://npmjs.com/advisories/782	tap>nyc>istanbul-lib-instrument>babel-generator>lodash	Y
install	lodash	moderate	npm install --save-dev tap@12.5.3	Prototype Pollution	https://npmjs.com/advisories/782	tap>nyc>istanbul-lib-instrument>babel-template>babel-traverse>babel-types>lodash	Y
install	lodash	moderate	npm install --save-dev tap@12.5.3	Prototype Pollution	https://npmjs.com/advisories/782	tap>nyc>istanbul-lib-instrument>babel-template>babel-traverse>lodash	Y
install	lodash	moderate	npm install --save-dev tap@12.5.3	Prototype Pollution	https://npmjs.com/advisories/782	tap>nyc>istanbul-lib-instrument>babel-template>babel-types>lodash	Y
install	lodash	moderate	npm install --save-dev tap@12.5.3	Prototype Pollution	https://npmjs.com/advisories/782	tap>nyc>istanbul-lib-instrument>babel-template>lodash	Y
install	lodash	moderate	npm install --save-dev tap@12.5.3	Prototype Pollution	https://npmjs.com/advisories/782	tap>nyc>istanbul-lib-instrument>babel-traverse>babel-types>lodash	Y
```
I am accountable to others for the "best-effort" basis on keeping packages secure to current standards. Fixing tap to 12.5.3 results in npm audit returning 0 known vulnerabilities.

Reply:
After tap@12.5.3
```
infintesimal@hishost:~/bw$ npm install --save-dev tap@12.5.3
+ tap@12.5.3
added 300 packages from 185 contributors and audited 580 packages in 10.058s
found 0 vulnerabilities

```

Reply:
I don’t think vulns matter in dev deps, generally. In this case, this prototype pollution attack only affects when deep merging untrusted code, which can’t happen here.

Reply:
The same goes for handlebars, which is only 'vulnerable' with untrusted templates. Browserify's dependencies only use trusted static templates.
Iirc, tunnel-agent is in the dependency tree but not used at all.

Generally, fixing these warnings can be worthwhile, even if only because they're annoying to the developer. In this case though, we're specifically using older versions because the newer versions don't support all our target platforms. If there is a a way to upgrade without losing compatibility, I'd be all for it.

Reply:
Agreed. I understand the dev deps is not relevant in a prod bundle. My security peers want to mitigate the risk wherever possible. I am looking at fresh conflicts in esm and the older node platforms the team supports. I will see if I can work something out. Right now build is breaking on esm.js. I'll be glad to have the PR accepted when I can prove that I can get through the build for all platforms. Cheers. Steve